src/vue/directives/tooltip.js

```diff
@@ -0,0 +1,387 @@
+      tooltipWrapper.style.left = '0'
+      tooltipWrapper.style.right = '0'
+      tooltipWrapper.style.margin = 'auto'
+      tooltipWrapper.style.transform = `translateY(${bottom})`
+      currentPosition = DIRECTIONS.top
+      refreshPositions()
+    }
+
+    // check if tw overflows left side
+    if (twRect.left < 0) {
+      tooltipWrapper.style.right = ''
+      tooltipWrapper.style.left = '50%'
+      refreshPositions()
+    }
+    // check if tw overflows right side
+    if (twRect.right > window.innerWidth) {
+      tooltipWrapper.style.left = '-50%'
+      refreshPositions()
+    }
+
+    // if tw still overflows, place it window center
+    if (twRect.left < 0 || twRect.right > window.innerWidth) {
+      const freeWindowSpace = (window.innerWidth - twRect.width) / 2
+      const centeredStartPoint = targetRect.left - freeWindowSpace
+      tooltipWrapper.style.left = -centeredStartPoint + 'px'
+    }
+
+    // if tw doesnt fit left or right sides
+    if (targetRect.width > spaceToLeft && targetRect.width > spaceToRight) {
+      if (currentPosition === DIRECTIONS.bottom) {
+        tooltipWrapper.style.left = '50%'
+        tooltipWrapper.style.transform = 'translate(-50%, 0)'
+      }
+      if (currentPosition === DIRECTIONS.top) {
+        tooltipWrapper.style.top = -(twRect.height + PADDING_SIZE * 2) + 'px'
+      }
+      refreshPositions()
+    }
+    // check if tw width wider than window
+    if (twRect.width > window.innerWidth) {
+      tooltipWrapper.style.margin = '0'
+      tooltipWrapper.style.width = appDiv.clientWidth + 'px'
+      tooltipWrapper.style.left = (0 - targetRect.left) + 'px'
+      refreshPositions()
+    }
+
+    // add animation to left or right directions
+    if (currentPosition === DIRECTIONS.left) {
+      let left = -(twRect.width + PADDING_SIZE) + 'px'
+      tooltipWrapper.style.transform = `translateX(${left})`
+      tooltipWrapper.style.left = '0'
+      tooltipWrapper.style.top = '0'
+    } else if (currentPosition === DIRECTIONS.right) {
+      let right = targetRect.width + PADDING_SIZE + 'px'
+      tooltipWrapper.style.transform = `translateX(${right})`
+      tooltipWrapper.style.left = '0'
+      tooltipWrapper.style.top = '0'
+    }
+    refreshPositions()
+    renderArrow(target)
+  }
+
+  function getPositionStyle (twRect) {
+    let positionStyle
+    switch (position) {
+      case DIRECTIONS.top:
+        positionStyle = {
+          top: targetHeight + MARGIN_SIZE,
+        }
+        break
+      case DIRECTIONS.right:
+        positionStyle = {
+          left: twRect.width + MARGIN_SIZE,
+        }
+        break
+      case DIRECTIONS.left:
+        positionStyle = {
+          left: 0 - twRect.width - MARGIN_SIZE,
+        }
+        break
+      case DIRECTIONS.bottom:
+      default:
+        positionStyle = {
+          top: 0 - targetHeight - MARGIN_SIZE,
+        }
+        break
+    }
+    return positionStyle
+  }
+
+  function renderTooltip (e, binding) {
```

Comment:
Render tooltip not inside the element, but inject in the document near the `#app`

Reply:
Done